reading_comprehension/qanet.py

```diff
@@ -0,0 +1,256 @@
+from typing import Any, Dict, List, Optional
+
+import torch
+from torch.nn.functional import nll_loss
+
+from allennlp.data import Vocabulary
+from allennlp.models.model import Model
+from allennlp.models.reading_comprehension.bidaf import BidirectionalAttentionFlow
+from allennlp.modules import Highway
+from allennlp.modules import Seq2SeqEncoder, TextFieldEmbedder
+from allennlp.modules.matrix_attention.matrix_attention import MatrixAttention
+from allennlp.nn import util, InitializerApplicator, RegularizerApplicator
+from allennlp.training.metrics import BooleanAccuracy, CategoricalAccuracy, SquadEmAndF1
+from reading_comprehension.utils import memory_effient_masked_softmax as masked_softmax
+
+
+@Model.register("qanet")
+class QANet(Model):
```

Comment:
Nit: we try to use [Google's definition of camel case](https://google.github.io/styleguide/javaguide.html#s5.3-camel-case), where this would be `QaNet`.  It makes it easier to read, so you don't have to guess where one acronym ends and another starts (imagine, e.g., `QALSTMNet` - much easier to parse if it's `QaLstmNet`).